### PR TITLE
feat: change base class 'react-dropdown-now'

### PR DESCRIPTION
**Pull request by iambumblehead**

### CODE REVIEW

**Comment by iambumblehead** [comment]
many changes needed yet

File: src/constants.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 export const DEFAULT_PLACEHOLDER_STRING = 'Select...';
 
 export const BASE_DEFAULT_PROPS = {
-  baseClassName: 'Dropdown',
+  baseClassName: 'react-dropdown-now',
```
[the remark this answers is not in the document]
Comment (lwhiteley) [reply src/constants.js:4]: also the styles will need to be updated .. maybe in the future changing to a css-in-js approach would be easier to maintain

https://github.com/iambumblehead/react-dropdown-now/blob/develop/style.css

[the remark this answers is not in the document]
Comment (iambumblehead) [reply src/constants.js:4]: @lwhiteley what classname do you think is best?

Locally one test shows this so far. I could change it to something else if you prefer,
```javascript
  t.is(
    component.html(),
    `<div class="react-dropdown-now">
       <div role="presentation" class="react-dropdown-now-control">
         <div class="react-dropdown-now-control-placeholder is-selected">one</div>
         <div class="react-dropdown-now-control-arrow">
           <span class="react-dropdown-now-control-arrow-icon"></span>
         </div>
       </div>
     </div>`
      .replace(/(\/n|\s*)(?=<)/gi, '')
  );
```

[the remark this answers is not in the document]
Comment (lwhiteley) [reply src/constants.js:4]: i think a short abbreviation would be better like `rdn` 

then others would be prefixed like

`rdn-control-arrow` etc.

**Comment by iambumblehead** [comment]
The arrow component will be a little bit smaller if we remove support for the custom arrow icon className. Instead, one could still define their own arrow icon and element with its own className.

https://github.com/iambumblehead/react-dropdown-now/blob/develop/src/components/Arrow.js#L21-L23

@lwhiteley do you think it is OK to remove support for the custom arrow className?

File: src/constants.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (iambumblehead) [reply src/constants.js:4]: if we want to keep the current classNames I'm okay with closing this MR. Changing the classnames will be a breaking change and will require updates to the stylesheet, the demo page and the README.

**Comment by lwhiteley** [comment]
> do you think it is OK to remove support for the custom arrow className?

personally i think the custom prefix is a little unnecessary and just causes extra complexity to maintain.
maybe should just decide on a definite prefix and not allow it to be changed by the library users.

This will ensure no "minor" breaking changes like this will occur again

**Comment by iambumblehead** [comment]
@lwhiteley thanks for being around and discussing changes with me

File: src/constants.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (lwhiteley) [reply src/constants.js:4]: im ok with it either way..

you can decide based on this comment: https://github.com/iambumblehead/react-dropdown-now/pull/71#issuecomment-716097884

[the remark this answers is not in the document]
Comment (iambumblehead) [reply src/constants.js:4]: @lwhiteley my internet connection is slow here so I didn't see your first response until after writing the last message... `rdn-control-arrow` seems good

**Comment by iambumblehead** [comment]
is-open,
is-selected,
is-disabled,
is-empty (menu)

**Comment by iambumblehead** [comment]
@lwhiteley please review!

File: src/Dropdown.js [hunk 2]
```diff
@@ -20,9 +20,6 @@ function Dropdown({
   onChange,
   onSelect,
   baseClassName,
-  controlClassName,
-  placeholderClassName,
-  menuClassName,
   arrowClassName,
```
Comment (lwhiteley) [inline src/Dropdown.js:23]: why keep the arrowClassName ?

why keep the baseClassname as a prop when the readme says its no longer configurable? maybe should jus be referenced as a constant

**Review by lwhiteley — COMMENTED** [review]
also need to update the typescript types

File: README.md [hunk 3]
```diff
@@ -129,13 +120,20 @@ More [examples in the docs folder.][2]
 - `onChange` always returns an object with aleast `{value, label}`
 - `option.type` is no longer needed to determine if the option is a group. Once the option has an `items` array then it is assumed to be a group.
 
+#### v3 => v4
+
+- removed configurable classNames `placeholderClassName`, `arrowClassName`, `menuClassName` and `controlClassName`
+- [changed classNames][10] to use className prefixing. stylesheets targeting the v3 component will need to be updated
+
+
```
Comment (lwhiteley) [inline README.md:128]: move this above the other set.. changelog should have the latest updates on top

**Comment by iambumblehead** [comment]
thanks for the review @lwhiteley I will update with those changes when there is a chance later today

**Comment by iambumblehead** [comment]
@lwhiteley please review :)

**Comment by iambumblehead** [comment]
the storybook addition is terrific

File: README.md [hunk 4]
```diff
@@ -98,16 +99,6 @@ const options = [{ id: 'custom-id', value: 1, label: 'awesome' }];
 
 ### Customizing
 
-#### Styling with classnames
```
Comment (lwhiteley) [inline README.md:101]: i think we should still list the class names that a dev should target for each part.

**Review by lwhiteley — COMMENTED** [review]
a part from the minor comment. looks good

File: README.md (hunk 4 above)
Comment (iambumblehead) [reply README.md:101]: the class names table is re added and updated

**Comment by iambumblehead** [comment]
@lwhiteley if there are any other issues feel free to let me know and I will try to resolve them

**Review by lwhiteley — APPROVED** [review]

**Comment by iambumblehead** [comment]
@lwhiteley should we merge this and then do `npm run release:major`?

**Comment by lwhiteley** [comment]
Yup but also ensure to rebuild the doc folder as well

**Comment by iambumblehead** [comment]
here it goes... after this pipeline completes

**Comment by iambumblehead** [comment]
I didn't have 'develop' checked out on the first try and that didn't work... so I did 'patch' release from the develop branch and that worked. Links in the README don't show the storybook.

**Pull request merged**